exec/java-exec/src/main/java/org/apache/drill/exec/cache/VectorAccessibleSerializable.java

```diff
@@ -146,36 +157,24 @@ public void writeToStream(OutputStream output) throws IOException {
     final DrillBuf[] incomingBuffers = batch.getBuffers();
     final UserBitShared.RecordBatchDef batchDef = batch.getDef();
 
-    /* DrillBuf associated with the selection vector */
-    DrillBuf svBuf = null;
-    Integer svCount =  null;
-
-    if (svMode == BatchSchema.SelectionVectorMode.TWO_BYTE) {
-      svCount = sv2.getCount();
-      svBuf = sv2.getBuffer(); //this calls retain() internally
-    }
-
     try {
       /* Write the metadata to the file */
       batchDef.writeDelimitedTo(output);
 
       /* If we have a selection vector, dump it to file first */
-      if (svBuf != null) {
-        allocator.write(svBuf, output);
-        sv2.setBuffer(svBuf);
-        svBuf.release(); // sv2 now owns the buffer
-        sv2.setRecordCount(svCount);
+      if (svMode == BatchSchema.SelectionVectorMode.TWO_BYTE) {
+        recordCount = sv2.getCount();
```

Comment:
Ultra minor comment: When SV2 is read, the recordCount comes from the batchDef; here it is taken from the sv2 itself (they should be the same anyway ....)